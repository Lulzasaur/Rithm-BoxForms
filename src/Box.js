import React, { Component } from 'react';

class Box extends Component {

  render() {
    const { height,width,color } = this.props;
    const styleObj = {
      backgroundColor: color,
      height:+height,
      width:+width,
      display: 'inline-block'
    }

    return (
      <div className="Box" style={styleObj}>
      <button onClick={this.props.removeMe}>X</button>
      </div>
    )
  }
}

export default Box;