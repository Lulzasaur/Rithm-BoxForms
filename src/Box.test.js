import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Box from './Box';

it('renders without crashing', function(){
    shallow(<Box/>)
})

it('matches snapshot', function(){
    let wrapper = shallow(<Box/>);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
})

it('did we make a black box', function(){
    let wrapper = shallow(<Box height={100} width={100} color='black'/>);
    expect(wrapper.props().style.backgroundColor).toEqual('black');
})

