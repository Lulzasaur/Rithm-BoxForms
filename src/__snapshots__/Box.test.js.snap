// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Box"
  style={
    Object {
      "backgroundColor": undefined,
      "display": "inline-block",
      "height": NaN,
      "width": NaN,
    }
  }
>
  <button>
    X
  </button>
</div>
`;
